serde/src/java/org/apache/hadoop/hive/serde2/typeinfo/TypeInfoUtils.java

```diff
@@ -1009,4 +1009,29 @@ public static boolean implicitConvertible(TypeInfo from, TypeInfo to) {
     }
     return false;
   }
+
+  public static boolean shouldEncloseQuotes(PrimitiveCategory category) {
```

Comment:
@kasakrisz does this look correct to you? I think you mentioned there is transformer for that

Reply:
I know transformer for identifiers but IIUC this is needed for literals.

Reply:
Literals are quoted by single quotes `'` but literals are quoted by double quotes (`"`) or backtick